android/KMEA/app/src/main/java/com/tavultesoft/kmea/KeyboardPickerActivity.java

```diff
@@ -362,6 +363,17 @@ protected static boolean addKeyboard(Context context, Keyboard keyboardInfo) {
 
       keyboardInfo.setNewKeyboard(true);
       KeyboardController.getInstance().add(keyboardInfo);
+      // Check if "other" keyboards of the same packageID and keyboardID need to update version
+      // Don't use forEach because we might be updating entries
+      for (int i=0; i<KeyboardController.getInstance().get().size(); i++) {
+        Keyboard otherKeyboard = KeyboardController.getInstance().getKeyboardInfo(i);
+        if (!otherKeyboard.equals(keyboardInfo) && otherKeyboard.getPackageID().equals(keyboardInfo.getPackageID())
+            && otherKeyboard.getKeyboardID().equals(keyboardInfo.getKeyboardID())
+            && (FileUtils.compareVersions(otherKeyboard.getVersion(), keyboardInfo.getVersion()) == FileUtils.VERSION_LOWER)) {
+          otherKeyboard.setVersion(keyboardInfo.getVersion());
+          KeyboardController.getInstance().set(i, otherKeyboard);
+        }
+      }
```

Comment:
I'm not understanding why the same keyboard+language may be installed more than once?

Reply:
Maybe `!otherKeyboard.equals(keyboardInfo)` is adding confusion.

We want to find matching packageID, matching keyboard ID, different languageID

The initial test setup results in
| packageID               | keyboardID                | language | version |
|-----------------------|------------------------|------------|---------|
sil_cameroon_qwerty | sil_cameroon_qwerty | aku-latn  | 6.0.4      |

Installing the keyboard package update originally resulted in
| packageID               | keyboardID                | language | version |
|-----------------------|------------------------|------------|---------|
|sil_cameroon_qwerty | sil_cameroon_qwerty | aku-latn  | 6.0.4     |
|sil_cameroon_qwerty | sil_cameroon_qwerty | aal-latn | 6.0.8      |

Reply:
So, it seems like the version information is being linked to the metadata entry rather than the package itself?